src/catalog.h

```diff
@@ -43,26 +44,44 @@ typedef enum CatalogTable
 	_MAX_CATALOG_TABLES,
 } CatalogTable;
 
+typedef struct TableInfoDef
+{
+	const char *schema_name;
+	const char *table_name;
+} TableInfoDef;
+
+typedef struct TableIndexDef
+{
+	int			length;
+	char	  **names;
+} TableIndexDef;
+
 #define INVALID_CATALOG_TABLE _MAX_CATALOG_TABLES
 #define INVALID_INDEXID -1
 
+#define CATALOG_PREFIX(catalog) \
+	(&catalog->prefix)
+#define CATALOG_TABLE_ID(catalog, tableid) \
```

Comment:
maybe this should be static inline `catalog_get_table_id`

Reply:
made `CATALOG_TABLE_ID` and `CATALOG_INDEX` both inlined.